api/swagger.yaml

```diff
@@ -0,0 +1,282 @@
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/base_message'
+      responses:
+        '201':
+          description: "message created"
+        default:
+          description: Unexpected error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/error'
+  /new-messages-long-polling:
+    get:
+      summary: |
+        Returns a list of new messages. New messages are messages created
+        after the last check for new messages. This endpoint is used for long polling.
+        It either returns a list of new messages or an empty body after a predefined time.
+      parameters:
+        - $ref: '#/components/parameters/user_id'
+      responses:
+        '200':
+          description: list of new messages
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/new_message_list_response'
+        default:
+          description: Unexpected error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/error'
+  /new-messages-polling:
+    get:
+      summary: |
+        Returns a list of new messages. New messages are messages created
+        after the last check for new messages. This endpoint is used for polling.
+        It either returns a list of new messages or an empty body immediately.
+      parameters:
+        - $ref: '#/components/parameters/user_id'
+      responses:
+        '200':
+          description: list of new messages
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/new_message_list_response'
+        default:
+          description: Unexpected error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/error'
+components:
+  parameters:
+    user_id:
+      name: user_id
+      in: query
+      description: User ID
+      required: true
+      schema:
+        type: string
+    chat_id:
+      name: chat_id
+      in: path
+      description: Chat ID
+      required: true
+      schema:
+        type: string
+    last_message_id:
+      name: last_message_id
+      in: query
+      description: Id of the last message already received. If set, the endpoint only returns newer messages
+      schema:
+        type: string
+  schemas:
+    chat_list_response:
+      type: object
+      properties:
+        chats:
+          type: array
+          items:
+            $ref: '#/components/schemas/detailed_chat'
+    chat_creation:
+      type: object
+      properties:
+        user_ids:
+          type: array
+          items:
+            type: string
+          example: [ 123, 321 ]
+        type:
+          $ref: '#/components/schemas/chat_type'
+    chat_response:
+      $ref: '#/components/schemas/detailed_chat'
```

Comment:
This would also include the message that could only be empty ;-)
No big deal.

Another thing:
should we not return a link to the new resource in the appropriate header?

Reply:
Not that my swagger for the dancer is correct in this way ;-)

Reply:
I think both ways are possible, returning the created resource and/or the link as header. And yes, some properties are empty or null after creation ;)